OpenSim/Common/Component.cpp

```diff
@@ -615,86 +615,27 @@ void Component::setParent(const Component& parent)
 
 std::string Component::getFullPathName() const
```

Comment:
@carmichaelong do you think it would be useful to have variants of these methods that give back a `ComponentPath`? In your experience of making these changes, can you think of a use case for such a variant? Just wondering. Perhaps it's best for us to leave out such variants for now, and add them in the future if they seem useful/necessary.